runtime/src/validate_block/implementation.rs

```diff
@@ -369,38 +382,151 @@ fn host_storage_read(key: &[u8], value_out: &mut [u8], value_offset: u32) -> Opt
+		.expect("Runing with a correct environment")
 }
 
 fn host_storage_append(key: &[u8], value: Vec<u8>) {
-	with_storage(|storage| storage.storage_append(key, value));
+	with_externalities(|ext| ext.storage_append(key.to_vec(), value))
+		.expect("Runing with a correct environment");
 }
 
 fn host_storage_next_key(key: &[u8]) -> Option<Vec<u8>> {
-	with_storage(|storage| storage.next_key(key))
+	with_externalities(|ext| ext.next_storage_key(key))
+		.expect("Runing with a correct environment")
+}
+
+fn host_storage_start_transaction() {
+	with_externalities(|ext| ext.storage_start_transaction())
+		.expect("Runing with a correct environment")
+}
+
+fn host_storage_rollback_transaction() {
+	with_externalities(|ext| ext.storage_rollback_transaction().ok())
+		.flatten()
+		.expect("Runing with a correct environment")
+}
+
+fn host_storage_commit_transaction() {
+	with_externalities(|ext| ext.storage_commit_transaction().ok())
+		.flatten()
+		.expect("Runing with a correct environment")
+}
+
+fn host_default_child_storage_get(storage_key: &[u8], key: &[u8]) -> Option<Vec<u8>> {
+	let child_info = ChildInfo::new_default(storage_key);
+	with_externalities(|ext| ext.child_storage(&child_info, key))
+		.expect("Runing with a correct environment")
+}
+
+fn host_default_child_storage_read(
+	storage_key: &[u8],
+	key: &[u8],
+	value_out: &mut [u8],
+	value_offset: u32,
+) -> Option<u32> {
+	let child_info = ChildInfo::new_default(storage_key);
+	match with_externalities(|ext| ext.child_storage(&child_info, key))
+		.expect("Runing with a correct environment") {
+		Some(value) => {
+			let value_offset = value_offset as usize;
+			let data = &value[value_offset.min(value.len())..];
+			let written = sp_std::cmp::min(data.len(), value_out.len());
+			value_out[..written].copy_from_slice(&data[..written]);
+			Some(value.len() as u32)
+		}
+		None => None,
+	}
+}
+
+fn host_default_child_storage_set(
+	storage_key: &[u8],
+	key: &[u8],
+	value: &[u8],
+) {
+	let child_info = ChildInfo::new_default(storage_key);
+	with_externalities(|ext| ext.place_child_storage(&child_info, key.to_vec(), Some(value.to_vec())))
+		.expect("Runing with a correct environment")
+}
+
+fn host_default_child_storage_clear(
+	storage_key: &[u8],
+	key: &[u8],
+) {
+	let child_info = ChildInfo::new_default(storage_key);
+	with_externalities(|ext| ext.place_child_storage(&child_info, key.to_vec(), None))
+		.expect("Runing with a correct environment")
+}
+
+fn host_default_child_storage_storage_kill(
+	storage_key: &[u8],
+) {
+	let child_info = ChildInfo::new_default(storage_key);
+	with_externalities(|ext| ext.kill_child_storage(&child_info))
+		.expect("Runing with a correct environment")
+}
+
+fn host_default_child_storage_exists(
+	storage_key: &[u8],
+	key: &[u8],
+) -> bool {
+	let child_info = ChildInfo::new_default(storage_key);
+	with_externalities(|ext| ext.exists_child_storage(&child_info, key))
+		.expect("Runing with a correct environment")
```

Comment:
Can we please put these `with_externalities` calls into a function that just panics directly. (so just a wrapper around it)

And please use a better expect message :D

Reply:
Done, not sure my new message is a lot better (the two new ones are copied from substrate).